lib/api/request_adapter.rb

```diff
@@ -99,6 +99,15 @@ def resource
       json_body.fetch("resource", json_body.except("action"))
     end
 
+    # A bulk request is of the form:
+    # {:action => :create, resources => [{},{}]}
+    # OR
+    # [{:action => :create, ...}, {:action => :delete, ...}]
+    # @returns true if this is a bulk request
+    def bulk?
+      !json_body.kind_of?(Hash) || json_body.key?("resources")
```

Comment:
may be easier to understand `json_body.kind_of?(Hash) && json_body.key?("resources")`

Reply:
@abellotti That's not the same thing...

`!A || B` -> `A && !B`

Reply:
But it is what we want though, a Hash that has a key resources.

Reply:
I was assuming the `!json_body.kind_of?(Hash)` was referring to the Array notation as listed in the comment.  That is I interpreted it as:

```
!json_body.kind_of?(Hash) || json_body.key?("resources")

Array || Hash-with-resources
```

Reply:
|json_body|bulk?|
|---|---|
|hash with :resources|true|
|hash without :resources|false|
|array of hashes|true|
|nil| don't care|
|some other class|don't care


I will make the successes more explicit rather than the vague `not`

Reply:
I may be missing something but I don't recall the second signature as bulk. @kbrock which API endpoint implements this ?

I know of the PATCH /api/:collection/:id with the similar signature as per https://manageiq.org/docs/reference/latest/api.overviiew/http.html example of PATCH /api/services/42, Array of attribute update specs, so really a single resource update, not bulk.